### ISSUE TITLE
bonus experiment variable

### ISSUE THREAD

**Issue by IanEisenberg** [expfactory/expfactory-docker#46]
Should be able to pay out "in proportion to" a value. So now just "if greater than", but the numeric value multiplied by some amount.

**Comment by vsoch** [comment]
In proportion to whata?

**Comment by IanEisenberg** [comment]
to the numeric value. So if you have 100 points I want to be able to pay 100*X, where X is some value. 

So I want to be able to turn "points" into payment.

**Comment by vsoch** [comment]
What points are you talking about? The experiment variable value?

**Comment by IanEisenberg** [comment]
Ah, yeah. Exactly. If someone defines an experiment variable that is of type "bonus" and is a "numeric" they should be able to say "take that value, multiply it by something and pay that out"

**Comment by vsoch** [comment]
If someone defines an experiment variable of type "Bonus" - the variable refers to what is collected during the experiment (eg, reaction time), I'm not sure how that would be multiplied to produce a monetary outcome? The use case of wanting to give some static reward for performance of the variable (eg, a fast reaction time) seems like the more common use case.

**Comment by IanEisenberg** [comment]
Can't believe I didn't see this response until now. People will definitely want what I am talking about. Two use cases:

(1) Bonus variable refers to points and each point represents 1cents. I'd want to pay them, literally, 1 cents per point.

(2) Bonus variable is determined by a choice (like in a delay discounting task). So bonus = one choice, which might be $500 or $1000, or $575, whatever the choice is. We'd want bonus pay to equal that amount * X, maybe .01. 

Those use cases are probably more likely than even rewarding for "fast" reaction times.

**Comment by vsoch** [comment]
Yes, right now it's up to the user to define the amount of bonus to allocate for some condition. You can pull out a calculator and do this manually, and entire the amount manually, it doesn't make sense to hard code into the infrastructure.

**Comment by IanEisenberg** [comment]
This isn't hard coded. It is a user specified ratio. The experiment determines the bonus variable (say points). So a subject completes the experiment and either gets 50 points or 79 points or whatever.

Individual researchers may want to bonus different amounts based on the points. Right now users can set different thresholds - if bonus (points) is above 55, pay $3, or whatever.

Another option people (including us) must have is to determine how those points translate to money in a continuous manner. I may want each point to be worth 1 cents, while someone else wants 25 cents.

So bonus paid will equal "bonus variable" (points) * [some multiplier].  The bonus variable is determined by the experiment template, but the [some multiplier] (and the option to do this kind of bonusing) should be determined by the user on expfactory.org

**Comment by IanEisenberg** [comment]
@rwblair
